packages/neon-domain/src/provider/NeoNS/main.ts

```diff
@@ -0,0 +1,25 @@
+import { logging, rpc, sc, u } from "@cityofzion/neon-core";
+
+export interface DomainProvider {
+  name: string;
+  resolveDomain(url: string, domain: string): Promise<string>;
+}
+const log = logging.default("neon-domain");
+
+export async function resolve(
+  url: string,
+  domain: string,
+  scriptHash: string, 
+  operation: string, 
+  args: any[]
+) {
+  const sb = new sc.ScriptBuilder();
+  const script = sb.emitAppCall(scriptHash, operation, args).str;
+  try {
+    const res = await rpc.Query.invokeScript(script).execute(url);
+    return rpc.StringParser(res.result.stack[0]);
+  } catch (err) {
+    log.error(`resolveDomain failed with : ${err.message}`);

```

Comment:
`resolveDomain` => `resolve`?

Reply:
Is there any benefit to this try/catch?  It looks like the only benefit is adding the function name that failed, but the error's stack trace will already provide that.

Reply:
removed resolve and put those three lines to resolveDomain. No the try/catch was not really necessary, removed it too.